File: solutions/swift/grains/3/Sources/Grains/Grains.swift
import Foundation

enum GrainsError: Error {
    case inputTooHigh(String)
    case inputTooLow(String)
}

private let minRange = 1
private let maxRange = 64

private func errorDescription(for number: Int) -> String {
    return "Input[\(number)] invalid. Input should be between \(minRange) and \(maxRange) (inclusive)"
}

func square(_ number: Int) throws -> UInt64 {
    guard number >= minRange else {
        throw GrainsError.inputTooLow(errorDescription(for: number))
    }
    guard number <= maxRange else {
        throw GrainsError.inputTooHigh(errorDescription(for: number))
    }
    return UInt64(pow(2.0, Double(number - 1)))
}

let total = UInt64.max

File: solutions/go/grains/1/grains.go
package grains

import (
	"fmt"
	"math"
)

func Square(number int) (uint64, error) {
	if number <= 0 || number > 64 {
		return 0, fmt.Errorf("Square data is out of bounds")
	}
	return uint64(math.Pow(2, float64(number-1))), nil
}

func Total() uint64 {
	return math.MaxUint64
}
